### PR TITLE
getting started docs for rclone

### PR DESCRIPTION
**Pull request by husky-parul**
Signed-off-by: Parul Singh <parsingh@redhat.com>

**Describe what this PR does**
Documentation with an example for using Rclone

**Is there anything that requires special attention?**

**Related issues:**
https://github.com/backube/scribe/issues/77

### CODE REVIEW

File: docs/usage/rclone/index.rst [hunk 1]
```diff
@@ -5,4 +5,189 @@ Rclone-based replication
...
    :hidden:
 
-Work in progress
+   database_example
+
+.. sidebar:: Contents
+
+   .. contents:: Rclone-based replication
+
+Rclone-based replication supports 1:1 asynchronous replication of volumes for use
```
Comment (JohnStrunk) [inline docs/usage/rclone/index.rst:14]: "1:many" or "1:n"

File: docs/usage/rclone/index.rst [hunk 2]
```diff
@@ -5,4 +5,189 @@ Rclone-based replication
...
+    sourcePVC: mysql-pv-claim
+    trigger:
+      schedule: "*/6 * * * *"
+     rclone:
+       copyMethod: Snapshot
+       capacity: 10Gi
+       accessModes: ["ReadWriteOnce"]
+
+Since the ``copyMethod`` specified above is ``Snapshot``, the Rclone data mover creates a ``VolumeSnapshot`` 
+of the source pvc ``mysql-pv-claim`` using the ``VolumeSnapshotClass`` provisioned by the CSI driver.
```
Comment (JohnStrunk) [inline docs/usage/rclone/index.rst:61]:
```suggestion
of the source pvc ``mysql-pv-claim`` using the cluster's default ``VolumeSnapshotClass``.
```

File: docs/usage/rclone/index.rst [hunk 3]
```diff
@@ -5,4 +5,189 @@ Rclone-based replication
...
+
+It then creates a temproray pvc ``scribe-src-database-source`` out of the VolumeSnapshot to transfer source
+data to the intermediary storage system like AWS S3 using the configurations provided in ``rclone-secret``
+
```
Comment (JohnStrunk) [inline docs/usage/rclone/index.rst:65]: How about adding some more information on the `rclone-secret` here (i.e., that it's just an rclone config file)?
A copy of the rclone.conf plus the command to load it as a secret `kubectl create secret generic ...blah...` would be good

**Comment by husky-parul** [comment]
@backube/scribemaintainers  This PR is ready for review.

File: docs/usage/rclone/index.rst [hunk 4]
```diff
@@ -5,4 +5,189 @@ Rclone-based replication
...
+   apiVersion: scribe/v1alpha1
+   kind: ReplicationSource
+   metadata:
+     name: database-source
+     namespace: source
+   spec:
+    sourcePVC: mysql-pv-claim
+    trigger:
+      schedule: "*/6 * * * *"
+     rclone:
```
Comment (JohnStrunk) [inline docs/usage/rclone/index.rst:55]: Does this need to have `rcloneDestPath` and `rcloneConfigSection` added?

File: docs/usage/rclone/index.rst [hunk 5]
```diff
@@ -5,4 +5,189 @@ Rclone-based replication
...
+        type:                  Reconciled
+      nextSyncTime:          2021-01-18T22:00:00Z
+
+
+In the above ``ReplicationSource`` object,
+
+- The Rclone configuations are provided via ``rclone-secret``
+- The PiT copy of the source data ``mysql-pv-claim`` will be created using ``VolumeSnapshot`` provisioned
+  by the CSI driver.
+- ``Rclone Dest Path`` indicates the location on the intermediary storage system where the source data
```
Comment (JohnStrunk) [inline docs/usage/rclone/index.rst:100]: `rcloneDestPath`

**Review by JohnStrunk — CHANGES_REQUESTED** [review]

File: docs/usage/rclone/index.rst (hunk 3 above)
Comment (husky-parul) [reply docs/usage/rclone/index.rst:65]: The command to create the secret is included in database_example.rst. I will explain the contents and probably include the link to Rclone AWS docs.

**Review by JohnStrunk — APPROVED** [review]

**Pull request merged**